pkg/tarball/write.go

```diff
@@ -143,7 +144,7 @@ func (c *Context) writeTar(ctx context.Context, tw *tar.Writer, fsys fs.FS, user
 			header.Devminor = int64(minor)
 		}
 		// work around some weirdness, without this we wind up with just the basename
-		header.Name = path
+		header.Name = filepath.Join(c.BaseDir, path)
```

Comment:
this _should_ be equivalent to `header.Name = path` when `c.BaseDir` is the zero value, but please correct me if I'm wrong!